test/node/byte_utils.test.ts

```diff
@@ -544,6 +566,32 @@ describe('ByteUtils', () => {
     });
   });
 
+  describe('randomBytes fallback case', () => {
+    let bsonWithNoCrypto;
+    before(function () {
+      bsonWithNoCrypto = loadBSONWithGlobal({
+        crypto: null,
+        // if we don't add a copy of Math here then we cannot spy on it for the test
+        Math: {
+          pow: Math.pow,
+          floor: Math.floor,
+          random: Math.random
+        }
+      });
+    });
+
+    after(function () {
+      sinon.restore();
+      bsonWithNoCrypto = null;
+    });
+
+    it('should fall back to Math.random implementation for random bytes if crypto is not present', () => {
+      const randomSpy = sinon.spy(bsonWithNoCrypto.Math, 'random');
+      new bsonWithNoCrypto.BSON.UUID();
+      expect(randomSpy).to.have.callCount(16);
```

Comment:
Why 16?  That's not entirely clear.  Is it not enough to simply assert that it was called, and that `crypto.randomBytes` was not called?

Reply:
Because 16 is the the length of a UUID. I would change it but `crypto` is null so it does not have a randomBytes method to not be called, so we cannot exactly assert that, any ideas?